crates/accelerate/src/sabre_layout.rs

```diff
@@ -47,14 +50,19 @@ pub fn sabre_layout_and_routing(
     };
     let seed_vec: Vec<u64> = outer_rng
         .sample_iter(&rand::distributions::Standard)
-        .take(num_layout_trials)
+        .take(num_layout_trials + partial_layouts.len())
         .collect();
     let dist = distance_matrix.as_array();
-    let res = if run_in_parallel && num_layout_trials > 1 {
+    let res = if run_in_parallel && (num_layout_trials > 1 || !partial_layouts.is_empty()) {
         seed_vec
             .into_par_iter()
             .enumerate()
             .map(|(index, seed_trial)| {
+                let partial = if index == num_layout_trials {
+                    Some(&partial_layouts[num_layout_trials - index])
+                } else {
+                    None
+                };
```

Comment:
Relatively minor style, but the API and logic might be clearer if we rename `num_layout_trials` to `num_random_trials`, then normalise both the random trials and `partial_layouts` into one single `partial_layouts` array at the start of the outer function - we can just use an empty vector for the full-random runs since they're a special case of "partial" that has _zero_ elements specified. Then we can unify the inner handling, and not need to worry about implicit indexing into either `num_layout_trials` or the `partial_layouts`, etc.

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/10721/commits/51bbcd3bbcbfab26e65e12ca3fdd17deeb32f6cf